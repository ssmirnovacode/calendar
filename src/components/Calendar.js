import { getMonthsToRender, getNextMonthYear, getPrevMonthYear, getShortDateString } from "../utils/helpers";
import Month from "./Month";
import '../sass/calendar.scss';
import { LocaleContext } from "../context/localeContext";
import { useEffect, useMemo, useState } from "react";

const Calendar = props => {

    const { numberOfMonths, arrows, startDate, endDate, onChange: setDates, locale='en-US' } = props;
    
    const initialDate = startDate ? startDate : new Date();
    const [ year, month, day ] = initialDate.toISOString().split('T')[0].split('-');

    const [ currentBox, setCurrentBox ] = useState({ month, year});

    const initialMonthsToRender = useMemo(() => getMonthsToRender(month, year, numberOfMonths), [month, year, numberOfMonths] ) ;

    const [ monthsToRender, setMonthsToRender ] = useState(initialMonthsToRender);

    useEffect(() => {
        //console.log(startDate)
    })

    const handleDaySelect = (e) => {
        const day = new Date(e.target.id);
        startDate && endDate ? setDates({ startDate: day, endDate: undefined}) :
        !startDate ? setDates(state => ({ startDate: day, endDate: state.endDate})) :
        setDates(state => ({ startDate: state.startDate, endDate: day}));
    }

    const handlePrevBtnClick = () => {
        const { month: prevMonth, year: prevYear } = getPrevMonthYear(currentBox.month, currentBox.year);
        setCurrentBox({ month: prevMonth, year: prevYear})
        setMonthsToRender(getMonthsToRender(prevMonth, prevYear, numberOfMonths));
    }

    const handleNextBtnClick = () => {
        const { month: nextMonth, year: nextYear } = getNextMonthYear(currentBox.month, currentBox.year);
        
        setCurrentBox({ month: nextMonth, year: nextYear})
        setMonthsToRender(getMonthsToRender(nextMonth, nextYear, numberOfMonths));
        
    }

    const renderArrows = () => {
        if (arrows) {
            return (
                <div className="calendar__arrows">
                    <button className="calendar__arrows--prev" onClick={handlePrevBtnClick}>&#8592;</button>
                    <button className="calendar__arrows--next" onClick={handleNextBtnClick}>&#8594;</button>
                </div>
            )
        }
    }

    return(
        <LocaleContext.Provider value={locale}>
            <div className="calendar">
                { renderArrows() }
                <div className="calendar__body">
                    {
                        monthsToRender.map(({ month, year }) => <Month 
                                                                    key={month.toString()+year.toString()+Date.now()} 
                                                                    month={month} 
                                                                    year={year} 
                                                                    handleDaySelect={handleDaySelect}
                                                                    startDate={getShortDateString(startDate)}
                                                                    endDate={getShortDateString(endDate)}
                                                                    />)
                    }
                </div>
            </div>
        </LocaleContext.Provider>
    )
}

export default Calendar;