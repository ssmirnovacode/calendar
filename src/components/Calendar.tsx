import { getMonthsToRender } from "../utils/helpers";
import "../css/calendar.css";
import React, { FC, useEffect, useMemo, useState } from "react";
import { initTheme } from "../utils/theme";
import { WARNING_BOTH_AVAIL_BLOCKED_DEFINED } from "../utils/constants";
import { Month, Year, assertCalendarProps } from "../types";
import { CalendarProps } from "../../calendar";
import { ArrowButtons, ArrowButtonsProps } from "./parcials/ArrowButtons";
import { DARK_THEME } from "../mockups/dark-theme";
import { CalendarContext } from "../context/calendarContext";
import { MonthContext } from "../context/monthContext";
import { MonthComponent } from "./MonthComponent";

const Calendar: FC<CalendarProps> = (props) => {
  const {
    numberOfMonths = 2,
    arrows = false,
    startDate,
    onChange: setDates,
    theme = DARK_THEME,
    clearDatesBtn = true,
    vertical = false,
    blockedDates,
    availableDates,
    disablePast = false,
  } = props;
  const initialDate: Date = startDate || new Date();

  const [, month, year]: [string, Month, Year] = initialDate
    .toLocaleDateString("es-ES")
    .split("/") as [string, Month, Year];

  const [currentBox, setCurrentBox] = useState<{ month: Month; year: Year }>({
    month,
    year,
  });

  const initialMonthsToRender = useMemo(
    () => getMonthsToRender(month, year, numberOfMonths),
    [month, year, numberOfMonths]
  );

  const [monthsToRender, setMonthsToRender] = useState<
    { month: Month; year: Year }[]
  >(initialMonthsToRender);

  useEffect(() => {
    assertCalendarProps(props);
    theme && initTheme({ ...theme, ...DARK_THEME }); // TODO create neutral default theme
    blockedDates &&
      availableDates &&
      console.warn(WARNING_BOTH_AVAIL_BLOCKED_DEFINED);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const arrowBtnsProps: ArrowButtonsProps = {
    disablePast,
    currentBox,
    numberOfMonths,
    setCurrentBox,
    setMonthsToRender,
  };

  return (
    <CalendarContext.Provider value={props}>
      <MonthContext.Provider value={{ month, year }}>
        <div className="calendar" data-testid="calendar">
          {arrows && !vertical ? <ArrowButtons {...arrowBtnsProps} /> : null}
          <div
            className={vertical ? "calendar__body vertical" : "calendar__body"}
          >
            {monthsToRender.map(({ month, year }) => (
              <MonthComponent
                key={month.toString() + year.toString() + Date.now()}
              />
            ))}
          </div>
          <div className="calendar__footer">
            {clearDatesBtn && (
              <button
                className="calendar__footer--btn"
                data-testid="clear-btn"
                onClick={() =>
                  setDates({ startDate: undefined, endDate: undefined })
                }
              >
                <span>&times;</span> Clear dates
              </button>
            )}
          </div>
        </div>
      </MonthContext.Provider>
    </CalendarContext.Provider>
  );
};

export default Calendar;
