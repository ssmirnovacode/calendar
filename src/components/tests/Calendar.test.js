import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Calendar from "../Calendar";

const START_DATE = new Date(2023, 8, 21); 
const NEXT_DATE = new Date(2023, 9, 21); 
const NEXT_MONTH = 11; // first month that wouldnt apper in 2-months view until the next btn click

describe('Calendar test', () => {
   
    const currentMonth = 9; // new Date().getMonth()+1;
    const currentDay = new Date().getDate();
    const requiredProps = { startDate: START_DATE, endDate: NEXT_DATE, onChange: ({startDate: START_DATE, endDate: NEXT_DATE}) => {} }

    test('default rendering', () => {
        render(<Calendar {...requiredProps} />); // overwise test will fail once month is passed
        const calendar = screen.getByTestId('calendar');
        expect(calendar).toBeInTheDocument();
        expect(calendar).toMatchSnapshot();
        const arrows = screen.queryByTestId('arrows');
        expect(arrows).toBeNull()
        const clearBtn = screen.queryByTestId('clear-btn');
        expect(clearBtn).toBeInTheDocument();
        const months = screen.getAllByTitle('month');
        expect(months.length).toBe(2)
    })

    test('custom months number', () => {
        render(<Calendar {...requiredProps}  numberOfMonths={8} />); 
        const months = screen.getAllByTitle('month');
        expect(months.length).toBe(8)
    })

    test('locale = DE', () => {
        render(<Calendar {...requiredProps} locale="de-DE" startDate={NEXT_DATE}/>); 
        const calendar = screen.getByTestId('calendar');
        expect(calendar).toMatchSnapshot();
    })

    test('vertical render', () => {
        render(<Calendar {...requiredProps} vertical startDate={START_DATE}/>); 
        const calendar = screen.getByTestId('calendar');
        expect(calendar).toMatchSnapshot();
    })

    test('disabled past', () => {
        render(<Calendar {...requiredProps} arrows disablePast />); 
        const calendar = screen.getByTestId('calendar');
        expect(calendar).toMatchSnapshot();
        const prevArrow = screen.queryByTestId('prev');
        expect(prevArrow).toBeNull();
        const nextArrow = screen.getByTestId('next');
        expect(nextArrow).toBeInTheDocument();   //toHaveClass
        for (let i=currentDay-1; i > 0; i--) {
            const pastDayCells = screen.getAllByTestId(`day-${currentDay-i}.${currentMonth}`); // empty cells corresponding to next and prev months included
            pastDayCells.forEach(cell => {
                expect(cell).toHaveClass('blocked')
            })
        }
    });

    test('blocked weekends', () => {
        render(<Calendar {...requiredProps} weekendsBlocked startDate={START_DATE}/>); 
        const saturday = screen.getByTestId('day-9.9');
        expect(saturday).toHaveClass('blocked')
        const sunday = screen.getByTestId('day-10.9');
        expect(sunday).toHaveClass('blocked')
    })

    test('with selected dates', () => {
        render(<Calendar {...requiredProps} startDate={START_DATE} endDate ={NEXT_DATE} />); 
        const calendar = screen.getByTestId('calendar');
        expect(calendar).toMatchSnapshot();
        const from = screen.getByTestId(`day-${START_DATE.getDate()-1}.${START_DATE.getMonth()+1}`); // -1 coz Date() returns 1 day less
        const between = screen.getByTestId(`day-${START_DATE.getDate()+5}.${START_DATE.getMonth()+1}`);
        const to = screen.getByTestId(`day-${NEXT_DATE.getDate()-1}.${NEXT_DATE.getMonth()+1}`);
        expect(from).toHaveClass('selected')
        expect(to).toHaveClass('selected')
        expect(between).toHaveClass('between')
    })

    test('render without clear btn', () => {
        render(<Calendar {...requiredProps} clearDatesBtn={false} startDate={START_DATE}/>); 
        const calendar = screen.getByTestId('calendar');
        expect(calendar).toMatchSnapshot();
        const clearBtn = screen.queryByTestId('clear-btn');
        expect(clearBtn).toBeNull()
    })

    test('render arrows', () => {
        render(<Calendar {...requiredProps} arrows startDate={START_DATE} />);
        const calendar = screen.getByTestId('calendar');
        expect(calendar).toMatchSnapshot();
        const arrows = screen.getByTestId('arrows');
        expect(arrows).toBeInTheDocument();
    })

    test('next btn click', () => {
        render(<Calendar {...requiredProps} arrows startDate={NEXT_DATE} />);
        const nextArrow = screen.getByTestId('next');
        userEvent.click(nextArrow);
        const nextMonth = screen.getByTestId(`month-${NEXT_MONTH+1}`); // +1 here and after is because data-testid is using real month numbers
        expect(nextMonth).toBeInTheDocument();
        const initialMonth = screen.queryByTestId(`month-${NEXT_DATE.getMonth()+1}`);       
        expect(initialMonth).toBeNull();
    })

    test('prev btn click', () => {
        render(<Calendar {...requiredProps} arrows startDate={NEXT_DATE} />);
        const prevArrow = screen.getByTestId('prev');
        userEvent.click(prevArrow);
        const initialMonth = screen.getByTestId(`month-${NEXT_DATE.getMonth()+1}`);       
        expect(initialMonth).toBeInTheDocument();
        const prevMonth = screen.getByTestId(`month-${NEXT_DATE.getMonth()}`);
        expect(prevMonth).toBeInTheDocument();
        const initialSecondMonth = screen.queryByTestId(`month-${NEXT_DATE.getMonth()+2
        }`);       
        expect(initialSecondMonth).toBeNull();
    })

})