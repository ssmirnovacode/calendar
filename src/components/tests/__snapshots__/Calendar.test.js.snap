// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar test default rendering 1`] = `
<div
  class="calendar"
  data-testid="calendar"
>
  <div
    class="calendar__body"
  >
    <div
      class="month"
      data-testid="month-9"
    >
      <div
        class="month__header"
      >
        <span>
          September
        </span>
         
        <span>
          2022
        </span>
      </div>
      <table
        class="table"
      >
        <thead
          class="table__header"
        >
          <tr>
            <th>
              Mon
            </th>
            <th>
              Tue
            </th>
            <th>
              Wed
            </th>
            <th>
              Thu
            </th>
            <th>
              Fri
            </th>
            <th>
              Sat
            </th>
            <th>
              Sun
            </th>
          </tr>
        </thead>
        <tbody
          class="table__body"
        >
          <tr>
            <td
              class="table__td hidden"
              id="2022-09-29"
            />
            <td
              class="table__td hidden"
              id="2022-09-30"
            />
            <td
              class="table__td hidden"
              id="2022-09-31"
            />
            <td
              class="table__td"
              id="2022-09-01"
            >
              1
            </td>
            <td
              class="table__td"
              id="2022-09-02"
            >
              2
            </td>
            <td
              class="table__td"
              id="2022-09-03"
            >
              3
            </td>
            <td
              class="table__td"
              id="2022-09-04"
            >
              4
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-09-05"
            >
              5
            </td>
            <td
              class="table__td"
              id="2022-09-06"
            >
              6
            </td>
            <td
              class="table__td"
              id="2022-09-07"
            >
              7
            </td>
            <td
              class="table__td"
              id="2022-09-08"
            >
              8
            </td>
            <td
              class="table__td"
              id="2022-09-09"
            >
              9
            </td>
            <td
              class="table__td"
              id="2022-09-10"
            >
              10
            </td>
            <td
              class="table__td"
              id="2022-09-11"
            >
              11
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-09-12"
            >
              12
            </td>
            <td
              class="table__td"
              id="2022-09-13"
            >
              13
            </td>
            <td
              class="table__td"
              id="2022-09-14"
            >
              14
            </td>
            <td
              class="table__td"
              id="2022-09-15"
            >
              15
            </td>
            <td
              class="table__td"
              id="2022-09-16"
            >
              16
            </td>
            <td
              class="table__td"
              id="2022-09-17"
            >
              17
            </td>
            <td
              class="table__td"
              id="2022-09-18"
            >
              18
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-09-19"
            >
              19
            </td>
            <td
              class="table__td selected"
              id="2022-09-20"
            >
              20
            </td>
            <td
              class="table__td"
              id="2022-09-21"
            >
              21
            </td>
            <td
              class="table__td"
              id="2022-09-22"
            >
              22
            </td>
            <td
              class="table__td"
              id="2022-09-23"
            >
              23
            </td>
            <td
              class="table__td"
              id="2022-09-24"
            >
              24
            </td>
            <td
              class="table__td"
              id="2022-09-25"
            >
              25
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-09-26"
            >
              26
            </td>
            <td
              class="table__td"
              id="2022-09-27"
            >
              27
            </td>
            <td
              class="table__td"
              id="2022-09-28"
            >
              28
            </td>
            <td
              class="table__td"
              id="2022-09-29"
            >
              29
            </td>
            <td
              class="table__td"
              id="2022-09-30"
            >
              30
            </td>
            <td
              class="table__td hidden"
              id="2022-09-01"
            />
            <td
              class="table__td hidden"
              id="2022-09-02"
            />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="month"
      data-testid="month-10"
    >
      <div
        class="month__header"
      >
        <span>
          October
        </span>
         
        <span>
          2022
        </span>
      </div>
      <table
        class="table"
      >
        <thead
          class="table__header"
        >
          <tr>
            <th>
              Mon
            </th>
            <th>
              Tue
            </th>
            <th>
              Wed
            </th>
            <th>
              Thu
            </th>
            <th>
              Fri
            </th>
            <th>
              Sat
            </th>
            <th>
              Sun
            </th>
          </tr>
        </thead>
        <tbody
          class="table__body"
        >
          <tr>
            <td
              class="table__td hidden"
              id="2022-10-26"
            />
            <td
              class="table__td hidden"
              id="2022-10-27"
            />
            <td
              class="table__td hidden"
              id="2022-10-28"
            />
            <td
              class="table__td hidden"
              id="2022-10-29"
            />
            <td
              class="table__td hidden"
              id="2022-10-30"
            />
            <td
              class="table__td"
              id="2022-10-01"
            >
              1
            </td>
            <td
              class="table__td"
              id="2022-10-02"
            >
              2
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-10-03"
            >
              3
            </td>
            <td
              class="table__td"
              id="2022-10-04"
            >
              4
            </td>
            <td
              class="table__td"
              id="2022-10-05"
            >
              5
            </td>
            <td
              class="table__td"
              id="2022-10-06"
            >
              6
            </td>
            <td
              class="table__td"
              id="2022-10-07"
            >
              7
            </td>
            <td
              class="table__td"
              id="2022-10-08"
            >
              8
            </td>
            <td
              class="table__td"
              id="2022-10-09"
            >
              9
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-10-10"
            >
              10
            </td>
            <td
              class="table__td"
              id="2022-10-11"
            >
              11
            </td>
            <td
              class="table__td"
              id="2022-10-12"
            >
              12
            </td>
            <td
              class="table__td"
              id="2022-10-13"
            >
              13
            </td>
            <td
              class="table__td"
              id="2022-10-14"
            >
              14
            </td>
            <td
              class="table__td"
              id="2022-10-15"
            >
              15
            </td>
            <td
              class="table__td"
              id="2022-10-16"
            >
              16
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-10-17"
            >
              17
            </td>
            <td
              class="table__td"
              id="2022-10-18"
            >
              18
            </td>
            <td
              class="table__td"
              id="2022-10-19"
            >
              19
            </td>
            <td
              class="table__td"
              id="2022-10-20"
            >
              20
            </td>
            <td
              class="table__td"
              id="2022-10-21"
            >
              21
            </td>
            <td
              class="table__td"
              id="2022-10-22"
            >
              22
            </td>
            <td
              class="table__td"
              id="2022-10-23"
            >
              23
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-10-24"
            >
              24
            </td>
            <td
              class="table__td"
              id="2022-10-25"
            >
              25
            </td>
            <td
              class="table__td"
              id="2022-10-26"
            >
              26
            </td>
            <td
              class="table__td"
              id="2022-10-27"
            >
              27
            </td>
            <td
              class="table__td"
              id="2022-10-28"
            >
              28
            </td>
            <td
              class="table__td"
              id="2022-10-29"
            >
              29
            </td>
            <td
              class="table__td"
              id="2022-10-30"
            >
              30
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-10-31"
            >
              31
            </td>
            <td
              class="table__td hidden"
              id="2022-10-01"
            />
            <td
              class="table__td hidden"
              id="2022-10-02"
            />
            <td
              class="table__td hidden"
              id="2022-10-03"
            />
            <td
              class="table__td hidden"
              id="2022-10-04"
            />
            <td
              class="table__td hidden"
              id="2022-10-05"
            />
            <td
              class="table__td hidden"
              id="2022-10-06"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    class="calendar__footer"
  >
    <button
      class="calendar__footer--btn"
    >
      <span>
        ×
      </span>
       Clear dates
    </button>
  </div>
</div>
`;

exports[`Calendar test render arrows 1`] = `
<div
  class="calendar"
  data-testid="calendar"
>
  <div
    class="calendar__arrows"
    data-testid="arrows"
  >
    <button
      class="calendar__arrows--prev"
      data-testid="prev"
    >
      ←
    </button>
    <button
      class="calendar__arrows--next"
      data-testid="next"
    >
      →
    </button>
  </div>
  <div
    class="calendar__body"
  >
    <div
      class="month"
      data-testid="month-9"
    >
      <div
        class="month__header"
      >
        <span>
          September
        </span>
         
        <span>
          2022
        </span>
      </div>
      <table
        class="table"
      >
        <thead
          class="table__header"
        >
          <tr>
            <th>
              Mon
            </th>
            <th>
              Tue
            </th>
            <th>
              Wed
            </th>
            <th>
              Thu
            </th>
            <th>
              Fri
            </th>
            <th>
              Sat
            </th>
            <th>
              Sun
            </th>
          </tr>
        </thead>
        <tbody
          class="table__body"
        >
          <tr>
            <td
              class="table__td hidden"
              id="2022-09-29"
            />
            <td
              class="table__td hidden"
              id="2022-09-30"
            />
            <td
              class="table__td hidden"
              id="2022-09-31"
            />
            <td
              class="table__td"
              id="2022-09-01"
            >
              1
            </td>
            <td
              class="table__td"
              id="2022-09-02"
            >
              2
            </td>
            <td
              class="table__td"
              id="2022-09-03"
            >
              3
            </td>
            <td
              class="table__td"
              id="2022-09-04"
            >
              4
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-09-05"
            >
              5
            </td>
            <td
              class="table__td"
              id="2022-09-06"
            >
              6
            </td>
            <td
              class="table__td"
              id="2022-09-07"
            >
              7
            </td>
            <td
              class="table__td"
              id="2022-09-08"
            >
              8
            </td>
            <td
              class="table__td"
              id="2022-09-09"
            >
              9
            </td>
            <td
              class="table__td"
              id="2022-09-10"
            >
              10
            </td>
            <td
              class="table__td"
              id="2022-09-11"
            >
              11
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-09-12"
            >
              12
            </td>
            <td
              class="table__td"
              id="2022-09-13"
            >
              13
            </td>
            <td
              class="table__td"
              id="2022-09-14"
            >
              14
            </td>
            <td
              class="table__td"
              id="2022-09-15"
            >
              15
            </td>
            <td
              class="table__td"
              id="2022-09-16"
            >
              16
            </td>
            <td
              class="table__td"
              id="2022-09-17"
            >
              17
            </td>
            <td
              class="table__td"
              id="2022-09-18"
            >
              18
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-09-19"
            >
              19
            </td>
            <td
              class="table__td selected"
              id="2022-09-20"
            >
              20
            </td>
            <td
              class="table__td"
              id="2022-09-21"
            >
              21
            </td>
            <td
              class="table__td"
              id="2022-09-22"
            >
              22
            </td>
            <td
              class="table__td"
              id="2022-09-23"
            >
              23
            </td>
            <td
              class="table__td"
              id="2022-09-24"
            >
              24
            </td>
            <td
              class="table__td"
              id="2022-09-25"
            >
              25
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-09-26"
            >
              26
            </td>
            <td
              class="table__td"
              id="2022-09-27"
            >
              27
            </td>
            <td
              class="table__td"
              id="2022-09-28"
            >
              28
            </td>
            <td
              class="table__td"
              id="2022-09-29"
            >
              29
            </td>
            <td
              class="table__td"
              id="2022-09-30"
            >
              30
            </td>
            <td
              class="table__td hidden"
              id="2022-09-01"
            />
            <td
              class="table__td hidden"
              id="2022-09-02"
            />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="month"
      data-testid="month-10"
    >
      <div
        class="month__header"
      >
        <span>
          October
        </span>
         
        <span>
          2022
        </span>
      </div>
      <table
        class="table"
      >
        <thead
          class="table__header"
        >
          <tr>
            <th>
              Mon
            </th>
            <th>
              Tue
            </th>
            <th>
              Wed
            </th>
            <th>
              Thu
            </th>
            <th>
              Fri
            </th>
            <th>
              Sat
            </th>
            <th>
              Sun
            </th>
          </tr>
        </thead>
        <tbody
          class="table__body"
        >
          <tr>
            <td
              class="table__td hidden"
              id="2022-10-26"
            />
            <td
              class="table__td hidden"
              id="2022-10-27"
            />
            <td
              class="table__td hidden"
              id="2022-10-28"
            />
            <td
              class="table__td hidden"
              id="2022-10-29"
            />
            <td
              class="table__td hidden"
              id="2022-10-30"
            />
            <td
              class="table__td"
              id="2022-10-01"
            >
              1
            </td>
            <td
              class="table__td"
              id="2022-10-02"
            >
              2
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-10-03"
            >
              3
            </td>
            <td
              class="table__td"
              id="2022-10-04"
            >
              4
            </td>
            <td
              class="table__td"
              id="2022-10-05"
            >
              5
            </td>
            <td
              class="table__td"
              id="2022-10-06"
            >
              6
            </td>
            <td
              class="table__td"
              id="2022-10-07"
            >
              7
            </td>
            <td
              class="table__td"
              id="2022-10-08"
            >
              8
            </td>
            <td
              class="table__td"
              id="2022-10-09"
            >
              9
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-10-10"
            >
              10
            </td>
            <td
              class="table__td"
              id="2022-10-11"
            >
              11
            </td>
            <td
              class="table__td"
              id="2022-10-12"
            >
              12
            </td>
            <td
              class="table__td"
              id="2022-10-13"
            >
              13
            </td>
            <td
              class="table__td"
              id="2022-10-14"
            >
              14
            </td>
            <td
              class="table__td"
              id="2022-10-15"
            >
              15
            </td>
            <td
              class="table__td"
              id="2022-10-16"
            >
              16
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-10-17"
            >
              17
            </td>
            <td
              class="table__td"
              id="2022-10-18"
            >
              18
            </td>
            <td
              class="table__td"
              id="2022-10-19"
            >
              19
            </td>
            <td
              class="table__td"
              id="2022-10-20"
            >
              20
            </td>
            <td
              class="table__td"
              id="2022-10-21"
            >
              21
            </td>
            <td
              class="table__td"
              id="2022-10-22"
            >
              22
            </td>
            <td
              class="table__td"
              id="2022-10-23"
            >
              23
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-10-24"
            >
              24
            </td>
            <td
              class="table__td"
              id="2022-10-25"
            >
              25
            </td>
            <td
              class="table__td"
              id="2022-10-26"
            >
              26
            </td>
            <td
              class="table__td"
              id="2022-10-27"
            >
              27
            </td>
            <td
              class="table__td"
              id="2022-10-28"
            >
              28
            </td>
            <td
              class="table__td"
              id="2022-10-29"
            >
              29
            </td>
            <td
              class="table__td"
              id="2022-10-30"
            >
              30
            </td>
          </tr>
          <tr>
            <td
              class="table__td"
              id="2022-10-31"
            >
              31
            </td>
            <td
              class="table__td hidden"
              id="2022-10-01"
            />
            <td
              class="table__td hidden"
              id="2022-10-02"
            />
            <td
              class="table__td hidden"
              id="2022-10-03"
            />
            <td
              class="table__td hidden"
              id="2022-10-04"
            />
            <td
              class="table__td hidden"
              id="2022-10-05"
            />
            <td
              class="table__td hidden"
              id="2022-10-06"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    class="calendar__footer"
  >
    <button
      class="calendar__footer--btn"
    >
      <span>
        ×
      </span>
       Clear dates
    </button>
  </div>
</div>
`;
